SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DefaultApplicationSettings.cs

```diff
@@ -137,5 +193,33 @@ private static Configuration OpenConfiguration()
             var executionFileMap = new ExeConfigurationFileMap { ExeConfigFilename = FilePath };
             return ConfigurationManager.OpenMappedExeConfiguration(executionFileMap, ConfigurationUserLevel.None);
         }
+
+        private void CreateConfigurationFileBackup()
+        {
+            try
+            {
+                configuration.SaveAs(BackupFilePath);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                AppCenterLog.Warn(AppCenterLog.LogTag, "Could not backup config file", e);
+            }
+        }
+
+        private bool RestoreConfigurationFile()
+        {
+            try
+            {
+                File.WriteAllText(FilePath, File.ReadAllText(BackupFilePath));
```

Comment:
How big files are? If this config is small, than its ok. Otherwise consider using streams

Reply:
I don't think that the configuration file is too big. 'File.Move' method already used fro configuration migration.